src/_igraph/convert.c

```diff
@@ -2606,11 +2676,120 @@ int igraphmodule_PyObject_to_vector_int_ptr_t(PyObject* list, igraph_vector_ptr_
     if (igraph_vector_ptr_push_back(vec, subvec)) {
       Py_DECREF(it);
       igraph_vector_int_destroy(subvec);
+      free(subvec);
       igraph_vector_ptr_destroy_all(vec);
       return 1;
     }
 
-    /* ownership of 'subvec' taken by 'vec' here */
+    /* ownership of 'subvec' taken by 'vec' here, we should still free the pointer */
```

Comment:
Same as above: there is no need to free the pointer _here_ as now it belongs to the `vector_ptr_t`. The code is correct, I am referring to the comment only.

Reply:
Agreed.